### PR TITLE
feat(comment)

### PR DESCRIPTION
**Pull request by sandymash**
- now users can comment on an inveastigation
- added ability to modify width and background color in button

### CODE REVIEW

File: client/src/components/App/Content/InvestigationForm/InvestigationInfo/InvestigatedPersonInfo/InvestigationMenu/CommentDialog/CommentDialog.tsx [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+    React.useEffect(resetInput, [comment]);
+
+    const handleInput = (event: React.ChangeEvent<HTMLTextAreaElement | HTMLInputElement>) => {
+        setCommentInput(event.target.value);
+    };
+
+    const sendComment = (commentToSend: string | null, errorMessage: string) => {
+        axios.post('/investigationInfo/comment', {comment: commentToSend, epidemiologyNumber})
+            .then(() => setComment(commentToSend))
```
Comment (daniel7764) [inline client/src/components/App/Content/InvestigationForm/InvestigationInfo/InvestigatedPersonInfo/InvestigationMenu/CommentDialog/CommentDialog.tsx:44]: add logs here

File: client/src/components/App/Content/InvestigationForm/InvestigationInfo/InvestigatedPersonInfo/InvestigationMenu/CommentDialog/CommentDialog.tsx [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+            <DialogContent className={classes.content}>
+                <TextField multiline fullWidth
+                           placeholder={COMMENT_PLACEHOLDER}
+                           value={commentInput} onChange={handleInput}
+                />
+            </DialogContent>
+            <DialogActions>
+                <PrimaryButton width='custom'
+                               disabled={!(Boolean(commentInput) && commentInput !== comment)}
```
Comment (daniel7764) [inline client/src/components/App/Content/InvestigationForm/InvestigationInfo/InvestigatedPersonInfo/InvestigationMenu/CommentDialog/CommentDialog.tsx:84]: Isn't !commentInput enough?

File: client/src/components/App/Content/InvestigationForm/InvestigationInfo/InvestigatedPersonInfo/InvestigationMenu/CommentDialog/CommentDialog.tsx [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+                               onClick={handleCommentSave}>
+                    {SAVE_BUTTON_TEXT}
+                </PrimaryButton>
+                <PrimaryButton width='custom' background='rgb(249, 89, 89)'
+                               disabled={!Boolean(comment)}
```
Comment (daniel7764) [inline client/src/components/App/Content/InvestigationForm/InvestigationInfo/InvestigatedPersonInfo/InvestigationMenu/CommentDialog/CommentDialog.tsx:89]: same here? maybe im wrong

File: client/src/components/App/Content/InvestigationForm/InvestigationInfo/InvestigatedPersonInfo/InvestigationMenu/InvestigationMenu.tsx [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+            <IconButton onClick={handleMenuOpen}>
+                <MoreVertIcon color='primary'/>
+            </IconButton>
+            <Menu classes={{paper: classes.menu}}
+                anchorEl={anchorEl}
+                keepMounted
+                open={open}
+                onClose={handleMenuClose}>
+                <MenuItem className={classes.menuItem} onClick={handleDialogOpen}>
+                    <CommentIcon htmlColor={Boolean(comment) ? '#91BF7C' : 'black'}/>
```
Comment (daniel7764) [inline client/src/components/App/Content/InvestigationForm/InvestigationInfo/InvestigatedPersonInfo/InvestigationMenu/InvestigationMenu.tsx:42]: put the hex value in a const

File: client/src/components/App/Content/InvestigationForm/InvestigationInfo/InvestigatedPersonInfo/InvestigationMenu/CommentDialog/CommentDialog.tsx (hunk 2 above)
Comment (sandymash) [reply client/src/components/App/Content/InvestigationForm/InvestigationInfo/InvestigatedPersonInfo/InvestigationMenu/CommentDialog/CommentDialog.tsx:84]: if they are both empty they would equal

File: client/src/components/App/Content/InvestigationForm/InvestigationInfo/InvestigatedPersonInfo/InvestigationMenu/CommentDialog/CommentDialog.tsx (hunk 3 above)
Comment (sandymash) [reply client/src/components/App/Content/InvestigationForm/InvestigationInfo/InvestigatedPersonInfo/InvestigationMenu/CommentDialog/CommentDialog.tsx:89]: i dont understand?

Comment (daniel7764) [reply client/src/components/App/Content/InvestigationForm/InvestigationInfo/InvestigatedPersonInfo/InvestigationMenu/CommentDialog/CommentDialog.tsx:89]: Isn't !comment enough here?

File: client/src/components/App/Content/InvestigationForm/InvestigationInfo/InvestigatedPersonInfo/InvestigatedPersonInfo.tsx [hunk 5]
```diff
@@ -18,6 +18,7 @@ import { setInvestigationStatus } from 'redux/Investigation/investigationActionC
 import useStyles from './InvestigatedPersonInfoStyles';
 import InfoItemWithIcon from './InfoItemWithIcon/InfoItemWithIcon';
 import useInvestigatedPersonInfo from './useInvestigatedPersonInfo';
+import InvestigationMenu from "./InvestigationMenu/InvestigationMenu";
```
Comment (roymaniv) [inline client/src/components/App/Content/InvestigationForm/InvestigationInfo/InvestigatedPersonInfo/InvestigatedPersonInfo.tsx:21]: single quotes

File: client/src/components/App/Content/InvestigationForm/InvestigationInfo/InvestigatedPersonInfo/InvestigationMenu/InvestigationMenuStyles.ts [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+const ITEM_HEIGHT = 48;
+
+const useStyles = makeStyles({
+    menu: {
+        maxHeight: ITEM_HEIGHT * 4.5,
+        width: '23ch',
+        borderRadius: '10px',
+        flip: false,
+        right: '2vw!important',
+        top: '12vh!important',
```
Comment (roymaniv) [inline client/src/components/App/Content/InvestigationForm/InvestigationInfo/InvestigatedPersonInfo/InvestigationMenu/InvestigationMenuStyles.ts:11]: has to be important?

File: client/src/components/App/Content/InvestigationForm/InvestigationInfo/InvestigationInfoBar.tsx [hunk 7]
```diff
@@ -19,6 +19,7 @@ import { setEpidemiologyNum, setLastOpenedEpidemiologyNum } from 'redux/Investig
 import useStyles from './InvestigationInfoBarStyles';
 import InvestigationMetadata from './InvestigationMetadata/InvestigationMetadata';
 import InvestigatedPersonInfo from './InvestigatedPersonInfo/InvestigatedPersonInfo';
+import {CommentContextProvider} from "./Context/CommentContext";
```
Comment (roymaniv) [inline client/src/components/App/Content/InvestigationForm/InvestigationInfo/InvestigationInfoBar.tsx:22]: single quotes

File: client/src/components/App/Content/InvestigationForm/InvestigationInfo/InvestigatedPersonInfo/InvestigationMenu/InvestigationMenuStyles.ts (hunk 6 above)
Comment (sandymash) [reply client/src/components/App/Content/InvestigationForm/InvestigationInfo/InvestigatedPersonInfo/InvestigationMenu/InvestigationMenuStyles.ts:11]: yup, or else its overriden by mui menu styles

**Review by roymaniv — DISMISSED** [review]

**Review by daniel7764 — APPROVED** [review]

**Review by daniel7764 — APPROVED** [review]

**Review by roymaniv — APPROVED** [review]

**Pull request merged**